cranelift/codegen/src/isa/mod.rs

```diff
@@ -378,6 +378,78 @@ impl<'a> dyn TargetIsa + 'a {
             pointer_width: self.pointer_width(),
         }
     }
+
+    /// Generate a `Capstone` context for disassembling bytecode for this architecture.
+    #[cfg(feature = "disas")]
+    pub fn to_capstone(&self) -> Result<capstone::Capstone, anyhow::Error> {
```

Comment:
I'd propose two changes to this method:
- Return `capstone::Error` instead of `anyhow::Error`, since this is Capstone-specific anyway, and let the caller map the error however they need to.
- Don't provide a default implementation (or provide one that returns `Err(capstone::Error::UnsupportedArch)`) and let the target-specific implementations figure out how to construct an appropriate `Capstone` object. This is slightly annoying since the implementations all need to be guarded with `#[cfg(feature = "disas")]` but I don't think that's a big deal.

Reply:
Much nicer, thanks!